src/rulesets-new/json-api-rules/resource-object-rules.ts

```diff
@@ -0,0 +1,323 @@
+import {
+  Ruleset,
+  RuleError,
+  ResponseRule,
+  ResponseBodyRule,
+  Matchers,
+} from "@useoptic/rulesets-base";
+import { links } from "../../docs";
+import { isOpenApiPath, isSingletonPath } from "../utils";
+
+const responseDataForPatch = new ResponseBodyRule({
+  name: "response data for patch",
+  matches: (responseBody, rulesContext) =>
+    rulesContext.operation.method === "patch" &&
+    responseBody.statusCode === "200" &&
+    responseBody.contentType === "application/vnd.api+json",
+  rule: (responseAssertions) => {
+    responseAssertions.body.added.matches({
+      schema: {
+        properties: {},
```

Comment:
We'll also want to follow up here... there's a bit more structure we'll want to require -- a JSON API resource.

Reply:
From what it seems like, there's a bit of overlap between the rules regarding shape. With the `matches` helper, this should be much easier to write a single rule